### ISSUE TITLE
Crash when playing songs "Memory Exhausted"

### ISSUE THREAD

**Issue by Skyran0** [ShadowMario/FNF-PsychEngine#10462]
### Describe your problem here.

I've tried to play a few songs on PsychEngine and all of them ended with the game crashing with the error "Memory Exhausted"

### Are you modding a build from source or with Lua?

Lua

### What is your build target?

Windows x64

**Comment by MemeHoovy** [comment]
> ### Describe your problem here.
> I've tried to play a few songs on PsychEngine and all of them ended with the game crashing with the error "Memory Exhausted"
> 
> ### Are you modding a build from source or with Lua?
> Lua
> 
> ### What is your build target?
> Windows x64
> 
> ### Did you edit anything in this build? If so, mention or summarize your changes.
> _No response_

does your computer or laptop have enough memory to run the game?
I suggest you open task manager and go to the performance tab and click on memory, then look at committed, that shows how much memory you used, and is available. If the first numbers on the left is just about the same as the number on the right,
ex. 29.2/31.9 (the max for me is 31.9, since I have 8gb of ram) then that means you are running out of memory, and should close some apps.
TL;DR: your pc or laptop might be out of memory, and you should close some apps

**Comment by Skyran0** [comment]
So I tried doing that, task manager shows 9.3/18.8 and when the game crashes 11.8/18.8, I currently have 16Gb of ram

**Comment by MemeHoovy** [comment]
> So I tried doing that, task manager shows 9.3/18.8 and when the game crashes 11.8/18.8, I currently have 16Gb of ram

huh, well I'm really just not sure, I never encountered that type of error at all, and you have plenty of memory available. So idk why it would say something like that, it could just be a bad ram stick or something, idk.

**Comment by Skyran0** [comment]
Yeah what really bothers me is that I played plenty of fnf mods which (I think) require better specs and it's really the first time I'm having this issue (also the first time I installed mods on the engine and not the engine with the mod already included)

**Comment by Itz-Miles** [comment]
in some builds, fnf can't work with more than 1 GIB of VRAM at a time. Your song could be too long, or there is too much cached to begin with.

**Comment by MemeHoovy** [comment]
> in some builds, fnf can't work with more than 1 GIB of VRAM at a time. Your song could be too long, or there is too much cached to begin with.

oh right, I forgot about vram

**Comment by Itz-Miles** [comment]
You might want to test builds with `<haxedef name="HXCPP_GC_BIG_BLOCKS">` enabled in project.xml. Otherwise, you'll have to go to the app's settings to determine how Windows handles it.
